Index: controllers/errors/go.mod
===================================================================
diff --git a/controllers/errors/go.mod b/controllers/errors/go.mod
deleted file mode 100644
--- a/controllers/errors/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,3 +0,0 @@
-module github.com/goRest/controllers/errors
-
-go 1.15
Index: controllers/go.mod
===================================================================
diff --git a/controllers/go.mod b/controllers/go.mod
deleted file mode 100644
--- a/controllers/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,10 +0,0 @@
-module github.com/lucas-gio/goRest/controllers
-
-go 1.15
-
-require (
-	github.com/gin-gonic/gin v1.6.3
-	go.mongodb.org/mongo-driver v1.4.6
-)
-
-//replace github.com/lucas-gio/goRest/configs => ./../configs
Index: routes/go.mod
===================================================================
diff --git a/routes/go.mod b/routes/go.mod
deleted file mode 100644
--- a/routes/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,10 +0,0 @@
-module github.com/lucas-gio/goRest/routes
-
-go 1.15
-
-require (
-	github.com/gin-gonic/gin v1.6.3
-	github.com/lucas-gio/goRest/controllers v0.0.0-00010101000000-000000000000
-)
-
-replace github.com/lucas-gio/goRest/controllers => ./../controllers
Index: configs/configuration/go.mod
===================================================================
diff --git a/configs/configuration/go.mod b/configs/configuration/go.mod
deleted file mode 100644
--- a/configs/configuration/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,8 +0,0 @@
-module github.com/lucas-gio/configs/configuration
-
-go 1.15
-
-require (
-	github.com/JeremyLoy/config v1.3.0
-	github.com/sirupsen/logrus v1.7.1
-)
Index: configs/go.mod
===================================================================
diff --git a/configs/go.mod b/configs/go.mod
deleted file mode 100644
--- a/configs/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,3 +0,0 @@
-module github.com/lucas-gio/goRest/configs
-
-go 1.15
Index: controllers/bicycles/go.mod
===================================================================
diff --git a/controllers/bicycles/go.mod b/controllers/bicycles/go.mod
deleted file mode 100644
--- a/controllers/bicycles/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,11 +0,0 @@
-module github.com/goRest/controllers/bicycles
-
-go 1.15
-
-replace github.com/lucas-gio/goRest/configs/datasource => ./../../configs/datasource
-
-require (
-	github.com/gin-gonic/gin v1.6.3
-	go.mongodb.org/mongo-driver v1.4.6
-	github.com/lucas-gio/goRest/configs/datasource v0.0.0
-)
Index: models/go.mod
===================================================================
diff --git a/models/go.mod b/models/go.mod
deleted file mode 100644
--- a/models/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,3 +0,0 @@
-module github.com/lucas-gio/goRest/models
-
-go 1.15
\ No newline at end of file
Index: configs/logger/go.mod
===================================================================
diff --git a/configs/logger/go.mod b/configs/logger/go.mod
deleted file mode 100644
--- a/configs/logger/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,5 +0,0 @@
-module github.com/lucas-gio/configs/logger
-
-go 1.15
-
-require github.com/sirupsen/logrus v1.7.1
Index: cmd/go.mod
===================================================================
diff --git a/cmd/go.mod b/cmd/go.mod
deleted file mode 100644
--- a/cmd/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,19 +0,0 @@
-module github.com/lucas-gio/goRest/cmd
-
-go 1.15
-
-replace github.com/lucas-gio/goRest/configs/configuration => ./../configs/configuration
-
-replace github.com/lucas-gio/goRest/configs/datasource => ./../configs/datasource
-
-replace github.com/lucas-gio/goRest/configs/logger => ./../configs/logger
-
-require (
-	github.com/gin-gonic/gin v1.6.3
-	github.com/lucas-gio/goRest/configs/configuration v0.0.0
-	github.com/lucas-gio/goRest/configs/datasource v0.0.0
-	github.com/lucas-gio/goRest/configs/logger v0.0.0
-	github.com/lucas-gio/goRest/routes v0.0.0-20210214025155-939cc15e12a1
-	github.com/sirupsen/logrus v1.7.1
-	go.mongodb.org/mongo-driver v1.4.6
-)
Index: configs/datasource/go.mod
===================================================================
diff --git a/configs/datasource/go.mod b/configs/datasource/go.mod
deleted file mode 100644
--- a/configs/datasource/go.mod	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ /dev/null	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
@@ -1,11 +0,0 @@
-module github.com/lucas-gio/configs/datasource
-
-go 1.15
-
-require (
-	github.com/lucas-gio/goRest/configs/configuration v0.0.0 // indirect
-	github.com/sirupsen/logrus v1.7.1
-	go.mongodb.org/mongo-driver v1.4.6
-)
-
-replace github.com/lucas-gio/goRest/configs/configuration => ./../configuration
Index: go.mod
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/go.mod b/go.mod
new file mode 100644
--- /dev/null	(date 1613844982049)
+++ b/go.mod	(date 1613844982049)
@@ -0,0 +1,9 @@
+module github.com/lucas-gio/goRest
+
+go 1.15
+
+require (
+	github.com/gin-gonic/gin v1.6.3
+	github.com/sirupsen/logrus v1.7.1 // indirect
+	go.mongodb.org/mongo-driver v1.4.6 // indirect
+)
\ No newline at end of file
Index: models/bicycle.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/models/bicycle.go b/internal/bicycles/bicycle.go
rename from models/bicycle.go
rename to internal/bicycles/bicycle.go
--- a/models/bicycle.go	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ b/internal/bicycles/bicycle.go	(date 1613845532135)
@@ -1,4 +1,4 @@
-package models
+package controllers
 
 type Bicycle struct {
 	Id          string
Index: routes/routes.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/routes/routes.go b/internal/routes/routes.go
rename from routes/routes.go
rename to internal/routes/routes.go
--- a/routes/routes.go	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ b/internal/routes/routes.go	(date 1613845627015)
@@ -1,13 +1,11 @@
 package routes
-/*
-import	"github.com/gin-gonic/gin"
-import "net/http"
-*/
-//"github.com/lucas-gio/goRest/controllers"
-//var bicyclesController BicyclesController = BicyclesController{}
 
-/*
-	router.GET("/404", func(c *gin.Context) {
+import (
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+router.GET("/404", func(c *gin.Context) {
 		c.HTML(http.StatusOK, "404.html", gin.H{"title": "Home Page"})
 	})
 
@@ -37,5 +35,5 @@
 		c.JSON(http.StatusOK, gin.H{
 			"message": "pong",
 		})
-	})*/
+	})
 }
Index: cmd/main.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cmd/main.go b/cmd/server/main.go
rename from cmd/main.go
rename to cmd/server/main.go
--- a/cmd/main.go	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ b/cmd/server/main.go	(date 1613845627007)
@@ -2,10 +2,10 @@
 
 import (
 	"github.com/gin-gonic/gin"
-	. "github.com/lucas-gio/goRest/configs/configuration"
-	. "github.com/lucas-gio/goRest/configs/datasource"
-	. "github.com/lucas-gio/goRest/configs/logger"
-	"github.com/lucas-gio/goRest/routes"
+	. "github.com/lucas-gio/goRest/internal/configs/configuration"
+	. "github.com/lucas-gio/goRest/internal/configs/datasource"
+	. "github.com/lucas-gio/goRest/internal/configs/logger"
+	"github.com/lucas-gio/goRest/internal/routes"
 )
 
 func main() {
Index: configs/configuration/configuration.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/configs/configuration/configuration.go b/internal/configs/configuration/configuration.go
rename from configs/configuration/configuration.go
rename to internal/configs/configuration/configuration.go
--- a/configs/configuration/configuration.go	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ b/internal/configs/configuration/configuration.go	(date 1613845627019)
@@ -1,7 +1,6 @@
 package configuration
 
 import (
-	"github.com/JeremyLoy/config"
 	log "github.com/sirupsen/logrus"
 	"path/filepath"
 	"sync"
Index: controllers/bicycles/bicyclesController.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/controllers/bicycles/bicyclesController.go b/internal/bicycles/bicyclesController.go
rename from controllers/bicycles/bicyclesController.go
rename to internal/bicycles/bicyclesController.go
--- a/controllers/bicycles/bicyclesController.go	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ b/internal/bicycles/bicyclesController.go	(date 1613845011271)
@@ -2,7 +2,6 @@
 
 import (
 	gin "github.com/gin-gonic/gin"
-	. "github.com/lucas-gio/goRest/configs/datasource"
 	"go.mongodb.org/mongo-driver/mongo"
 
 	"net/http"
Index: configs/datasource/datasource.go
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/configs/datasource/datasource.go b/internal/configs/datasource/datasource.go
rename from configs/datasource/datasource.go
rename to internal/configs/datasource/datasource.go
--- a/configs/datasource/datasource.go	(revision a03daf53c47fb419677c4b9f9900ae8eebd56e8a)
+++ b/internal/configs/datasource/datasource.go	(date 1613845627011)
@@ -2,7 +2,6 @@
 
 import (
 	"context"
-	. "github.com/lucas-gio/goRest/configs/configuration"
 	log "github.com/sirupsen/logrus"
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
@@ -31,7 +30,7 @@
 	return &this.ctx
 }
 
-func (this *Datasource) ConnectToMongodb(configuration *Configuration) {
+func (this *Datasource) ConnectToMongodb(configuration *configuration.Configuration) {
 	onceDatasource.Do(func() {
 		this.ctx, this.cancel = context.WithTimeout(context.Background(), 10*time.Second)
 
diff --git a/deployments/Dockerfile b/deployments/Dockerfile
new file mode 100644
diff --git a/dbScripts/scripts.js b/dbTestScripts/scripts.js
rename from dbScripts/scripts.js
rename to dbTestScripts/scripts.js
diff --git a/controllers/errors/errorsController.go b/internal/errors/errorsController.go
rename from controllers/errors/errorsController.go
rename to internal/errors/errorsController.go
diff --git a/configs/logger/logger.go b/internal/configs/logger/logger.go
rename from configs/logger/logger.go
rename to internal/configs/logger/logger.go
